components/automate-deployment/pkg/manifest/parser/parser.go

```diff
@@ -70,7 +68,7 @@ func parseA2Manifest(body []byte) (*manifest.A2, error) {
 	a2 := &manifest.A2{}
 	err := json.Unmarshal(body, a2)
 	if err != nil {
-		return nil, errors.Wrap(err, "Failed to A2 manifest")
+		return nil, manifest.ErrCannotParse
```

Comment:
I generally like error constants like this, but here we are losing most of the helpful error information. It is a _lot_more verbose, but if you want be able to check for a specific error and not lose detail information, we might consider making a custom error. For instance, over in the certauthority package we do:

https://github.com/chef/automate/blob/master/components/automate-deployment/pkg/certauthority/certauthority.go#L141-L152

which we can then check specifically using a type check:

https://github.com/chef/automate/blob/master/components/automate-deployment/pkg/certauthority/certauthority_test.go#L231-L232

Reply:
I'll give you your parse errors :dango: